nmma/em/likelihood.py

```diff
@@ -144,20 +140,72 @@ def log_likelihood(self):
+                time_array = np.round(np.linspace(self.tmin, self.tmax, n), 2)
+                yaml_filters = yaml_dict["config"]["withTime"]["filters"]
+                systematicsprior.validate_filters(yaml_filters)
+
+                for filter_group in yaml_filters:
+                    if isinstance(filter_group, list):
+                        filt = "___".join(filter_group)
+                    elif filter_group is None:
+                        filt = "all"
+                    else:
+                        filt = filter_group
+
+                    globals()[f"sys_err_{filt}_array"] = np.array([])
+
+                    for i in range(1, n + 1):
+                        value = self.parameters.get(f"sys_err_{filt}{i}")
+                        globals()[f"sys_err_{filt}_array"] = np.append(globals()[f"sys_err_{filt}_array"], value)
+
+                for filter_group in yaml_dict["config"]["withTime"]["filters"]:
+                    if isinstance(filter_group, list):
+                        filt = "___".join(filter_group)
+                    elif filter_group is None:
+                        filt = "all"
+                    else:
+                        filt = filter_group
+                    globals()[f"sys_err_{filt}_interped"] = interp1d(time_array, globals()[f"sys_err_{filt}_array"], fill_value="extrapolate", bounds_error=False)
+
         # compare the estimated light curve and the measured data
         minus_chisquare_total = 0.0
         gaussprob_total = 0.0
+
         for filt in self.filters:
             # decompose the data
             data_time = self.light_curve_data[filt][:, 0]
             data_mag = self.light_curve_data[filt][:, 1]
             data_sigma = self.light_curve_data[filt][:, 2]
 
-            # include the error budget into calculation
-            if 'em_syserr' in self.parameters:
-                data_sigma = np.sqrt(data_sigma**2 + self.parameters['em_syserr']**2)
+            if self.systematics_file is not None:
+                if yaml_dict["config"]["withTime"]["value"]:
+                    yaml_filters = yaml_dict["config"]["withTime"]["filters"]
+
+                    filter_group_finite_idx_match = False
+
+                    if yaml_filters is not None and yaml_filters != []:
+                        for yaml_filt in yaml_filters:
+                            if yaml_filt is not None and filt in yaml_filt:
+                                if len(yaml_filt) == 1:
+                                    filters_to_use = yaml_filt[0]
+                                else:
+                                    filters_to_use = "___".join(yaml_filt)
+                                filter_group_finite_idx_match = True
+                                break
+                    if not filter_group_finite_idx_match:
+                        filters_to_use = "all"
+                    data_sigma = np.sqrt(data_sigma**2 + (globals()[f"sys_err_{filters_to_use}_interped"](data_time)) ** 2)
```

Comment:
why `global` here?

Reply:
Using globals in the `all` case as well just to homogenise all the dynamically allocated variable calls

Reply:
Instead of `globals` another way could be monkey patching, similar to what is being done for the new `from_list` method of `PriorDict`